implementation/endpoints/src/tcp_server_endpoint_impl.cpp

```diff
@@ -272,6 +272,63 @@ void tcp_server_endpoint_impl::accept_cbk(const connection::ptr& _connection,
                 VSOMEIP_WARNING << "tcp_server_endpoint::connect: couldn't enable "
                         << "keep_alive: " << its_error.message();
             }
+
+            // Enable sending of keep-alive messages on connection-oriented sockets
+            // Enables (nonzero) or disables (zero) the periodic transmission of messages
+            // Should the connected party fail to respond to these messages, the connection is considered broken
+            int optval = 1;
+            int rc;
+            rc = setsockopt(new_connection_socket.native_handle(), SOL_SOCKET, SO_KEEPALIVE, &optval, sizeof(optval));
+            if (rc != 0) {
+                VSOMEIP_WARNING << "tcp_server_endpoint::connect: couldn't enable keep_alive(10)" ;
+            }
+#ifndef __QNX__
```

Comment:
Hey @joeyoravec 

Thanks for this Pull Request.

One suggestion to maybe avoid "#ifdef platform" and the hardcoded values, the tcp keep alive options could be added as new configurations in the vsomeip json files. 
If no configurations would be present, the default from the platform would be used (this would be preferable for linux based systems, where sysctl is used)
If configurations for the keep alive would be present, those would be used instead, for all platforms.